driver-core/src/main/java/com/datastax/driver/core/ControlConnection.java

```diff
@@ -647,6 +651,19 @@ private static void refreshNodeListAndTokenMap(Connection connection, Cluster.Ma
             cluster.metadata.rebuildTokenMap(partitioner, tokenMap);
     }
 
+    private static boolean isValidPeer(Row peerRow, InetSocketAddress addr) {
+        return peerRow.getColumnDefinitions().contains("host_id")
+                && peerRow.getColumnDefinitions().contains("rack")
+                && peerRow.getColumnDefinitions().contains("tokens")
+                && addr != null
```

Comment:
Small nit: the `addr` parameter doesn't seem to belong here; imo the code would be a bit more readable if this method only cared about `peerRow`; the if block above could then be `if (addr == null || !isValidPeer(row))`